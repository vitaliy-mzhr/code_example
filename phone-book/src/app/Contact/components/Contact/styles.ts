import styled from 'styled-components';

const ContactStyles = styled.div`
  margin: .5rem 0;
  
  .card-action {
    width: 100%;
  }
  .phones-container {
    display: flex;
    justify-content: flex-start;
    flex-direction: column;
    align-items: flex-start;
    flex-wrap: nowrap;
    padding: .5rem .5rem .5rem 1rem;
    
    .phone {
      display: flex;
      justify-content: flex-start;
      align-items: center;
      text-decoration: none;
      color: ${props => props.theme.colors.mainColor}
    }
  }
  .phones-header {
    padding: .5rem 1.5rem;
    font-weight: 600;
  }
  .buttons-container {
    position: relative;
    display: flex;
    justify-content: flex-end;
    align-items: center;
  }
  .contact-content {
    display: none;
    padding: 0;
    &.expanded {
      display: block;
    }
    &:last-child {
      padding-bottom: 0;
    }
  }
`;

export default ContactStyles;
