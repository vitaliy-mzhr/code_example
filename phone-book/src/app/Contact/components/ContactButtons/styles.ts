import styled from 'styled-components';

const ContactButtonStyles = styled.div`
  padding: .25rem;
  .contact-button {
    
    .contact-icon {
      font-size: 1.5rem;
    }
    
    &:hover {
      .edit-icon {
        color: ${props => props.theme.colors.blueColor};
      }
      .delete-icon {
        color: ${props => props.theme.colors.redColor};
      }
      .close-icon {
        color: ${props => props.theme.colors.mainColor};
      }
    }
  }
`;

export default ContactButtonStyles;
