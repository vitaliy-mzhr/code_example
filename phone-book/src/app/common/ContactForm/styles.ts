import styled from 'styled-components';

const ContactFormStyles = styled.div`
  h3 {
    text-align: center;
  }
  .form-container {
    margin: 0 auto;
    padding-top: 2rem;
    max-width: 20rem;
    display: flex;
    flex-direction: column;
    justify-content: center;
  }
  .buttons-container {
    display: flex;
    justify-content: center;
    padding: 1rem;
    button {
      margin: 0 .25rem;
    }
  }
`;

export default ContactFormStyles;
