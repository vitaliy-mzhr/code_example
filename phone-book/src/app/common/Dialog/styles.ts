import styled from 'styled-components';

const AlertDialogStyles = styled.div`
  min-width: 14rem;
  .dialog-title {
    h2 {
      text-align: center;
    }
  }
  .buttons-container {
    display: flex;
    justify-content: space-evenly;
    align-items: center;
  }
  
  .confirm-button-icon {
    color: ${props => props.theme.colors.greenColor};
  }
  .cancel-button-icon {
    color: ${props => props.theme.colors.errorColor};
  }
`;

export default AlertDialogStyles;
