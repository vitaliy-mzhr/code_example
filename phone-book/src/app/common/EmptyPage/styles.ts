import styled from 'styled-components';

const EmptyPageStyles = styled.div`
  display: flex;
  justify-content: center;
  width: 100%;
  .page-message-container {
    margin-top: 2rem;
    padding: 2rem;
}
`;

export default EmptyPageStyles;
