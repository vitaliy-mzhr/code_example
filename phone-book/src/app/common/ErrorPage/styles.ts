import styled from 'styled-components';

const ErrorPageStyles = styled.div`
  display: flex;
  justify-content: center;
  width: 100%;
  .page-message-container {
    margin-top: 2rem;
    padding: 2rem;
    color: ${props => props.theme.colors.errorColor};
}
`;

export default ErrorPageStyles;
