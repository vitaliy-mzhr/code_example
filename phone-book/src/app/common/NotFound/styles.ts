import styled from 'styled-components';

const NotFoundStyles = styled.div`
  display: flex;
  justify-content: center;
  align-items: center;
  flex-direction: column;
  
  a {
    text-decoration: none;
  }
`;

export default NotFoundStyles;
