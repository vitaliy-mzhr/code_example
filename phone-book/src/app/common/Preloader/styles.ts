import styled from 'styled-components';

const PreloaderStyles = styled.div`
  position: absolute;
  width: 100%;
  height: calc(100% - ${props => props.theme.dimensions.navMinHeightRem}rem);
  display: flex;
  justify-content: center;
  align-items: center;
  
  @media all and (max-width: ${props => props.theme.media.mobileMaxWidthPx}px) {
   height: calc(100% - ${props => props.theme.dimensions.navMinHeightRem * 0.9}rem);
  }
`;

export default PreloaderStyles;