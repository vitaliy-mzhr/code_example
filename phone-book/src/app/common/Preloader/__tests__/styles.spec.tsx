import * as React from 'react';
import PreloaderStyles from '../styles';
import { shallow } from 'enzyme';
import 'jest-styled-components';
import { mainTheme } from '../../../../styles/themes';

it('PreloadertStyles applied correctly', () => {
  expect(shallow(<PreloaderStyles theme={mainTheme} />)).toMatchSnapshot();
});