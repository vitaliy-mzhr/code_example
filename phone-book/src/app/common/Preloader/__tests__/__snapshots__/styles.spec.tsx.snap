// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PreloadertStyles applied correctly 1`] = `
.c0 {
  position: absolute;
  width: 100%;
  height: calc(100% - 4.5rem);
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

@media all and (max-width:768px) {
  .c0 {
    height: calc(100% - 4.05rem);
  }
}

<div
  className="c0"
/>
`;
